lib/fs.js

```diff
@@ -1615,6 +1615,11 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (typeof options === 'string')
+    options = { encoding: options };
+  else if (options && typeof options !== 'object')
+    throw new TypeError('Bad arguments');
```

Comment:
Other functions throws same error message... 

BUT, If anyone have a nice idea, I would like to take more general and kind error message.